### PR TITLE
WIP design experiments for e-money (refactors)

### PR DESCRIPTION
**Pull request by jbibla**
Closes #ISSUE

**Description:**

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by jbibla** [comment]
WIP
Screen Shot 2020-03-11 at 10 55 28 AM

File: src/components/common/CurrencySelector.vue [hunk 1]
```diff
@@ -0,0 +1,75 @@
+<template>
+  <div v-if="visible" class="currency-selector">
```
Comment (faboweb) [inline src/components/common/CurrencySelector.vue]: handle outside of component with v-if?

File: src/components/common/CurrencySelector.vue [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    }
+  },
+  data() {
+    return {
+      selectedCurrency: `EUR`,
+      supportedCurrencies: [`EUR`, `USD`, `GBP`, `CHF`, `JPY`]
+    }
+  },
+  //   mounted() {
+  //     const localCurrency = localStorage.getItem(
```
Comment (faboweb) [inline src/components/common/CurrencySelector.vue]: move to store

File: src/components/common/CurrencySelector.vue [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  //       `selectedCurrency`,
+  //       this.selectedCurrency
+  //     )
+  //     if (localCurrency) this.selectedCurrency = localCurrency
+  //   },
+  methods: {
+    setCurrency() {
+      localStorage.setItem(`selectedCurrency`, this.selectedCurrency)
```
Comment (faboweb) [inline src/components/common/CurrencySelector.vue]: move to store

File: src/components/common/TmBalance.vue [hunk 4]
```diff
@@ -1,230 +1,104 @@
...
-      <div class="button-container">
-        <TmBtn
-          class="send-button"
-          value="Send"
-          type="secondary"
-          @click.native="onSend()"
+
+      <ul v-if="overview && !overview.totalStakeFiatValue">
+        <BalanceRow
+          :crypto-amount="overview.totalStake"
```
Comment (faboweb) [inline src/components/common/TmBalance.vue]: i see, this should be queried in the balance query

**Comment by faboweb** [comment]
Looks good so far. Obviously did too many improvements on other parts of the app

File: src/components/common/TmBalance.vue (hunk 4 above)
Comment (jbibla) [reply src/components/common/TmBalance.vue]: `total` can be queried through balances but `available` cannot

Comment (jbibla) [reply src/components/common/TmBalance.vue]: also, would be nice if `rewards` were included in the balances query

Comment (faboweb) [reply src/components/common/TmBalance.vue]: sure, but isn't that just putting everything into one query? better to separate the concerns don't you think?

File: src/components/common/CurrencySelector.vue (hunk 1 above)
Comment (jbibla) [reply src/components/common/CurrencySelector.vue]: it's a prop

Comment (faboweb) [reply src/components/common/CurrencySelector.vue]: I suggest handling this outside via a v-if. doesn't make sense to pass in a prop and then do the v-if inside

**Comment by faboweb** [comment]
need better icons imo

**Comment by faboweb** [comment]
comparing with designs:
designs have the token name with the rewards
designs have fiat balance in line
designs have captions per button

**Pull request closed without merging**